### PR TITLE
add zk & marathon

### PR DESCRIPTION
**Pull request by skonto**
- add zk
- add marathon
- modify submits and dispatcher creation calls to use zk
- update run_tests.sh script (just to make it run from there... need proper arg handling this was pending from previous commits)
- update run.sh script
- update docs

### CODE REVIEW

**Comment by skyluc** [comment]
Can you also update the generated `index.html` file, to contain links to zookeeper and marathon if launched?

**Comment by skyluc** [comment]
`spark.zk.uri` should be added to `application.conf.template`.

**Comment by skonto** [comment]
Zk is not the default master so its optional and its uri is generated only when need and added to the file no reason to stay around empty i guess...

**Comment by skonto** [comment]
I will update the index.html file.

**Comment by skonto** [comment]
I think its ok now check pls.

File: mesos-docker/run/run.sh [hunk 1]
```diff
@@ -190,9 +196,17 @@ function start_master {
...
+  if [[ -n $INSTALL_ZK ]]; then
+    zk="--zk=zk://$dip:2181/mesos"
+  fi
+
+  start_master_command="/usr/sbin/mesos-master $zk --ip=$dip $(quote_if_non_empty $MESOS_MASTER_CONFIG)"
   start_master_command="$(get_mesos_update_command) ; $start_master_command"
 
+
+  #upmesos="apt-get update -o Dir::Etc::sourcelist=sources.list.d/mesosphere.list -o Dir::Etc::sourceparts=- -o APT::Get::List-Cleanup=0"
+  #start_master_command="$upmesos ; apt-get install --only-upgrade mesos; $start_master_command"
```
Comment (dragos) [inline mesos-docker/run/run.sh]: Are these comments still needed here? Generally we don't commit commented out code, since there's Git for history.

**Comment by dragos** [comment]
One small style observation, otherwise LGTM.

**Comment by skonto** [comment]
yes i was about to delete them... merge left overs... i will  fix it

**Comment by skonto** [comment]
done... should i merge it?

**Comment by dragos** [comment]
go for it!

**Pull request merged**